lib/features/common/domain/entity/operation/result.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:flutter_template/features/common/domain/entity/operation/failure.dart';
+
+/// {@template result.class}
+/// Result
+///
+/// [Result.ok] - Named constructor for successful Result - [ResultOk]
+///
+/// [Result.failed] - Named constructor for Result with error - [ResultFailed]
+/// {@endtemplate}
+sealed class Result<TData, TErr extends Failure> {
```

Comment:
I like everything, but I'm not a fan of the placement of this class. Why is it located in features/common/domain/entity? It looks like a universal base class. Maybe we should move it out of the features and into a conditional folder like lib/core/result? What do you think about this?